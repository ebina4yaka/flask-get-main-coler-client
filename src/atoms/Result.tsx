import { ReactElement } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { createStyles } from '@material-ui/core'
import Paper from '@material-ui/core/Paper'

type Props = {
  colors: string[] | undefined
}

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      padding: theme.spacing(1),
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      textAlign: 'center',
    },
    color: {
      margin: theme.spacing(1),
      padding: theme.spacing(3, 2, 3),
      width: '14%',
    },
  })
)

export default function Result(props: Props): ReactElement {
  const { colors } = props
  if (!colors) {
    return <></>
  }
  const classes = useStyles()
  return (
    <Paper className={classes.root}>
      {colors.map((color) => {
        return (
          <div
            key={color}
            className={classes.color}
            style={{ backgroundColor: color }}
          />
        )
      })}
    </Paper>
  )
}
