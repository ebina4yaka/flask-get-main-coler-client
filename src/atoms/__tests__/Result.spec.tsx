import renderer from 'react-test-renderer'
import Result from '../Result'

test('Snapshot test for Result', () => {
  const tree = renderer.create(<Result colors={['#fff', '#555']} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot test for Result color empty', () => {
  const tree = renderer.create(<Result colors={undefined} />).toJSON()
  expect(tree).toMatchSnapshot()
})
