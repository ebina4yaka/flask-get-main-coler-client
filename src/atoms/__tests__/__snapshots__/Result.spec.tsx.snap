// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Result 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="makeStyles-color-2"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  />
  <div
    className="makeStyles-color-2"
    style={
      Object {
        "backgroundColor": "#555",
      }
    }
  />
</div>
`;

exports[`Snapshot test for Result color empty 1`] = `null`;
